File: QA.Engine.Administration.WebApp.Core/ClientApp/models/PageViewModel.ts

module Models {

    // $Classes/Enums/Interfaces(filter)[template][separator]
    // filter (optional): Matches the name or full name of the current item. * = match any, wrap in [] to match attributes or prefix with : to match interfaces or base classes.
    // template: The template to repeat for each matched item
    // separator (optional): A separator template that is placed between all templates e.g. $Properties[public $name: $Type][, ]
    
    // More info: http://frhagn.github.io/Typewriter/
    
    
    export class PageViewModel {
        
        // ID
        public id: number = 0;
        // ISARCHIVE
        public isArchive: boolean = false;
        // PARENTID
        public parentId: number = null;
        // ALIAS
        public alias: string = null;
        // TITLE
        public title: string = null;
        // ZONENAME
        public zoneName: string = null;
        // EXTENSIONID
        public extensionId: number = null;
        // INDEXORDER
        public indexOrder: number = null;
        // ISVISIBLE
        public isVisible: boolean = null;
        // VERSIONOFID
        public versionOfId: number = null;
        // PUBLISHED
        public published: boolean = false;
        // ISINSITEMAP
        public isInSiteMap: boolean = null;
        // DISCRIMINATOR
        public discriminator: string = null;
        // DISCRIMINATORID
        public discriminatorId: number = 0;
        // DISCRIMINATORTITLE
        public discriminatorTitle: string = null;
        // ICONURL
        public iconUrl: string = null;
        // WIDGETS
        public widgets: WidgetViewModel[] = [];
        // CHILDREN
        public children: PageViewModel[] = [];
        // CONTENTVERSIONS
        public contentVersions: PageViewModel[] = [];
        // REGIONS
        public regions: RegionViewModel[] = [];
        // HASWIDGETS
        public hasWidgets: boolean = false;
        // HASCHILDREN
        public hasChildren: boolean = false;
        // HASCONTENTVERSION
        public hasContentVersion: boolean = false;
        // HASREGIONS
        public hasRegions: boolean = false;
    }
}

File: QA.Engine.Administration.WebApp.Core/Models/PageViewModel.cs
using System.Collections.Generic;
using System.Linq;
using QA.Engine.Administration.WebApp.Core.Annotations;

namespace QA.Engine.Administration.WebApp.Core.Models
{
    [TypeScriptType]
    public class PageViewModel
    {
        public int Id { get; set; }
        public bool IsArchive { get; set; }
        public int? ParentId { get; set; }
        public string Alias { get; set; }
        public string Title { get; set; }
        public string ZoneName { get; set; }
        public int? ExtensionId { get; set; }
        public int? IndexOrder { get; set; }
        public bool? IsVisible { get; set; }
        public int? VersionOfId { get; set; }
        public bool Published { get; set; }
        public bool? IsInSiteMap { get; set; }
        public string Discriminator { get; set; }
        public int DiscriminatorId { get; set; }
        public string DiscriminatorTitle { get; set; }
        public string IconUrl { get; set; }
        public List<WidgetViewModel> Widgets { get; set; }
        public List<PageViewModel> Children { get; set; }
        public List<PageViewModel> ContentVersions { get; set; }
        public List<RegionViewModel> Regions { get; set; }
        public bool HasWidgets { get { return Widgets?.Any() ?? false; } }
        public bool HasChildren { get { return Children?.Any() ?? false; } }
        public bool HasContentVersion { get { return ContentVersions?.Any() ?? false; } }
        public bool HasRegions { get { return Regions?.Any() ?? false; } }
    }
}
